File: src/LoginServlet.java
import javax.servlet.*;
import javax.servlet.http.*;
import javax.servlet.annotation.*;
import java.io.IOException;
import java.io.StringWriter;
import java.io.Writer;

@WebServlet(name = "LoginServlet", value = "/LoginServlet")
public class LoginServlet extends HttpServlet {
    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {

    }

    @Override
    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
        String name = request.getParameter("name");
        String pwd = request.getParameter("pwd");
        System.out.println("name:" + name);
        System.out.println("pwd:" + pwd);
        if (name == null || "".equals(name)) {
            System.out.println("用户名不为空");
            return;
        }
        if (pwd == null || "".equals(pwd)) {
            System.out.println("密码不为空");
            return;
        }
        if ("admin".equals(name) && "123".equals(pwd)) {
            System.out.println("登录成功");
            // 重定向
            response.sendRedirect("/index.jsp");
            // 请求转发
            request.getRequestDispatcher("/index.jsp").forward(request, response);
        } else {
            Writer out = response.getWriter();
            out.write("登录失败");
        }
    }
}

File: src/servlet/LoginServlet.kt
package servlet

import service.UserService
import service.impl.UserServiceImpl
import java.io.IOException
import javax.servlet.ServletException
import javax.servlet.annotation.WebServlet
import javax.servlet.http.HttpServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@WebServlet(name = "servlet.LoginServlet", value = ["/servlet.LoginServlet"])
class LoginServlet : HttpServlet() {
    @Throws(ServletException::class, IOException::class)
    override fun doGet(request: HttpServletRequest, response: HttpServletResponse) {
    }

    @Throws(ServletException::class, IOException::class)
    override fun doPost(request: HttpServletRequest, response: HttpServletResponse) {
        request.characterEncoding = "utf-8"
        response.characterEncoding = "utf-8"
        response.contentType = "text/html;charset=utf-8"
        val name = request.getParameter("name")
        val pwd = request.getParameter("pwd")
        val userService: UserService = UserServiceImpl()
        val user = userService.login(name, pwd)
        println("登录成功，ID:${user.id} UserName:${user.userName} Password:${user.pwd}")
        if (user.userName != "") {
            response.sendRedirect(request.contextPath + "/index.jsp")
        } else {
            response.sendRedirect(request.contextPath + "/login.jsp?message=login_error")
        }
    }
}